### PR TITLE
Hook calls can now be forked into a worker queue entry

### PR DESCRIPTION
**Pull request by annando**
This helps isolating the separated calls so that a crash cannot cause much problems.

Before this we had some similar coding for the "cron" hook. This was custom made for this single hook. The new solution is now universal and is used for the hook "delivery_normal" as well that is responsible for delivering content to third party networks.

Failures in the addon had previously caused endless postings. This now can't happen anymore.

### CODE REVIEW

File: src/Core/Addon.php [hunk 1]
```diff
@@ -185,6 +186,25 @@ public static function loadHooks()
...
 
+	/**
+	 * @brief Forks a hook.
+	 *
+	 * Use this function when you want to fork a hook via the worker.
+	 *
+	 * @param string $name of the hook to call
+	 * @param string|array $data to transmit to the callback handler
+	 */
+	public static function ForkHooks($priority, $name, $data = null)
```
Comment (MrPetovan) [inline src/Core/Addon.php]: Guess what? Right idea, wrong capitalization: it should be `forkHooks`.

Comment (annando) [reply src/Core/Addon.php]: One day I will learn ...

Comment (MrPetovan) [reply src/Core/Addon.php]: `ClassName`
`methodName`
`$variable_name`

Comment (annando) [reply src/Core/Addon.php]: Done.

**Pull request merged**